Orange/widgets/data/owselectbydataindex.py

```diff
@@ -89,6 +108,10 @@ def commit(self):
             matching_output = self.data[row_sel]
             non_matching_output = self.data[~row_sel]
             annotated_output = create_annotated_table(self.data, row_sel)
+
+        summary = len(matching_output) if matching_output else self.info.NoOutput
```

Comment:
This is not true. 
`matching_output` evaluates to False when `len(matching_output) == 0`, causing `summary = self.info.NoOutput`, even though dataset with 0 instances was send to the output.